sofagym/envs/Maze/MazeToolbox.py

```diff
@@ -273,10 +272,18 @@ def getReward(rootNode):
         done, reward
 
     """
+    done = False
+
+    goal = rootNode.Goal.GoalMO.position.value[0]
+    ball_pos = rootNode.Simulation.Sphere.sphere_mo.position.value[0]
+    dist = np.linalg.norm(ball_pos - goal)
 
     reward, cost = rootNode.Reward.getReward()
 
-    return False, reward
+    if dist < 5 or ball_pos[1] < -100:
```

Comment:
It's not a good practice to have [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). I recommend to name them to give them a meaning.